src/server/tiered_storage.cc

```diff
@@ -124,6 +128,22 @@ class TieredStorage::ShardOpManager : public tiering::OpManager {
     return false;
   }
 
+  // Load all values from bin by their hashes
+  void Defragment(tiering::DiskSegment segment, string_view value) {
+    auto hashes = ts_->bins_->DeleteBin(segment, value);
+    for (auto [dbid, hash] : hashes) {
+      auto it = db_slice_->GetDBTable(dbid)->prime.Find(hash);
```

Comment:
who implemented find by hash value in dash table?
I can see this code in main branch..

Reply:
Nobody 🙂 It doesn't compile with what I pushed. Generally, we can't find a value just by hash, but we can loop over all values with the same hash and try finding a value that is external and lies on the given segment

Reply:
yes that is what yue did in a PR, you will also need to check that the external value offset is the same offset of the hash in the segment this way you can be sure its the same key